server/client_server.go

```diff
@@ -22,10 +24,125 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
+type IssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values,omitempty"`
+}
+
+type FieldInfo struct {
+	Values []interface{} `json:"values,omitempty"`
+}
+
+type FieldValues struct {
+	FieldID string `json:"fieldId,omitempty"`
+}
+
+type FieldId struct {
+	Values []FieldValues `json:"values,omitempty"`
+}
+
+type Version struct {
+	VersionInfo string `json:"version,omitempty"`
+}
+
+const (
+	CreateMetaAPIEndpoint = "rest/api/2/issue/createmeta/"
+	PivotVersion          = "8.4.0"
+)
+
 // GetCreateMeta returns the metadata needed to implement the UI and validation of
 // creating new Jira issues.
 func (client jiraServerClient) GetCreateMeta(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, error) {
-	cimd, resp, err := client.Jira.Issue.GetCreateMetaWithOptions(options)
+	ServerInfoApiEndpoint := "rest/api/2/serverInfo"
+	v := new(Version)
+	req, err := client.Jira.NewRequest(http.MethodGet, ServerInfoApiEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = client.Jira.Do(req, v)
+	if err != nil {
+		return nil, err
+	}
+
+	currentVersion, err := version.NewVersion(v.VersionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	pivotVersion, err := version.NewVersion(PivotVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *jira.CreateMetaInfo
+	var resp *jira.Response
+	if currentVersion.LessThan(pivotVersion) {
+		cimd, response, error := client.Jira.Issue.GetCreateMetaWithOptions(options)
+		info = cimd
+		resp = response
+		err = error
+	} else {
+		cd, response, err := client.Jira.Project.ListWithOptions(options)
+		meta := new(jira.CreateMetaInfo)
+
+		if err == nil {
+			for i := 0; i < len(*cd); i++ {
+				meta.Expand = (*cd)[i].Expand
+				apiEndpoint := fmt.Sprintf("%s%s/issuetypes", CreateMetaAPIEndpoint, (*cd)[i].ID)
+				req, err := client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
+				if err != nil {
```

Comment:
check this one as well here you are creating a new variable named err if it does not nill you are breaking from the loop but you are not updating the error not able to understand this